src/main/java/software/amazon/awssdk/trustedidentitypropagation/TrustedIdentityPropagationPlugin.java

```diff
@@ -96,6 +96,12 @@ private TrustedIdentityPropagationPlugin(Builder builder) {
         Validate.notNull(builder.accessRoleArn, "Access Role Arn must be provided.");
         Validate.notNull(builder.webTokenProvider, "ID token supplier must be provided.");
 
+        if (builder.ssoOidcClient == null) {
```

Comment:
Something I'd like to see us fix in a V2 is that we are overriding the STS client's credential provider with _whatever_ is on the OIDC client.  
Can you update this to:
1. Only share the credentials provider if the STS client was not provided
2. Accept an optional credentials provider directly, instead of laundering it through the oidcClient